apps/ledger-live-mobile/src/newArch/features/FirmwareUpdate/components/UpdateBanner/UpdateBanner.test.tsx

```diff
@@ -0,0 +1,399 @@
+  ...jest.requireActual("@react-navigation/native"),
+  useRoute: jest.fn().mockReturnValue({ params: {} }),
+  useNavigation: jest.fn().mockReturnValue({ navigate: jest.fn() }),
+}));
+
+// Mock useLatestFirmware
+jest.mock("@ledgerhq/live-common/device/hooks/useLatestFirmware", () => ({
+  useLatestFirmware: jest.fn(),
+}));
+const { useLatestFirmware } = jest.requireMock(
+  "@ledgerhq/live-common/device/hooks/useLatestFirmware",
+);
+
+// Mock navigation to new and old update flows
+jest.mock("../../utils/navigateToNewUpdateFlow", () => ({
+  navigateToNewUpdateFlow: jest.fn(),
+}));
+jest.mock("../../utils/navigateToOldUpdateFlow", () => ({
+  navigateToOldUpdateFlow: jest.fn(),
+}));
+const { navigateToNewUpdateFlow } = jest.requireMock("../../utils/navigateToNewUpdateFlow");
+const { navigateToOldUpdateFlow } = jest.requireMock("../../utils/navigateToOldUpdateFlow");
+
+const oldUpdateFlowNotSupportedDataSet: Array<{
+  deviceModelId: DeviceModelId;
+  version: string;
+  productName: string;
+}> = [
+  { deviceModelId: DeviceModelId.nanoS, version: "1.6.0", productName: "Nano S" },
+  { deviceModelId: DeviceModelId.nanoX, version: "1.2.9", productName: "Nano X" },
+  { deviceModelId: DeviceModelId.nanoSP, version: "0.9.9", productName: "Nano S Plus" },
+];
+
+const oldUpdateFlowSupportedDataSet: Array<{
+  deviceModelId: DeviceModelId;
+  version: string;
+  productName: string;
+}> = [
+  { deviceModelId: DeviceModelId.nanoS, version: "1.6.1", productName: "Nano S" },
+  { deviceModelId: DeviceModelId.nanoX, version: "1.3.0", productName: "Nano X" },
+  { deviceModelId: DeviceModelId.nanoSP, version: "1.0.0", productName: "Nano S Plus" },
+];
+
+const newUpdateFlowSupportedDataSet: Array<{
+  deviceModelId: DeviceModelId;
+  version: string;
+  productName: string;
+}> = [
+  { deviceModelId: DeviceModelId.stax, version: "1.0.0", productName: "Stax" },
+  { deviceModelId: DeviceModelId.europa, version: "1.0.0", productName: "Europa" },
+];
+
+describe("<UpdateBanner />", () => {
+  let PlatformSpy: jest.SpyInstance;
+  beforeEach(() => {
+    jest.restoreAllMocks();
+    jest.resetAllMocks();
+    PlatformSpy = jest.spyOn(ReactNative, "Platform", "get");
+  });
+
+  it("should not display the firmware update banner if there is no update", async () => {
+    useLatestFirmware.mockReturnValue(null);
+
+    const mockDeviceModelId = DeviceModelId.nanoS;
+    const mockDeviceVersion = "2.0.0";
+    render(<UpdateBanner onBackFromUpdate={() => {}} />, {
+      overrideInitialState: (state: State) => ({
+        ...state,
+        settings: {
+          ...state.settings,
+          ...makeMockSettings({
+            modelId: mockDeviceModelId,
+            version: mockDeviceVersion,
+            hasCompletedOnboarding: true,
+            wired: true,
+          }),
+        },
+        appstate: {
+          ...state.appstate,
+          hasConnectedDevice: true,
+        },
+      }),
+    });
```

Comment:
Same here you can create a function that you reuse everywhere with only what you need to update as parameter